src/cryptenroll/cryptenroll.c

```diff
@@ -253,7 +260,29 @@ static int parse_argv(int argc, char *argv[]) {
                         r = parse_path_argument(optarg, /* suppress_root= */ true, &arg_unlock_keyfile);
                         if (r < 0)
                                 return r;
+
+                        arg_unlock_type = UNLOCK_KEYFILE;
+                        break;
+
+                case ARG_UNLOCK_FIDO2_DEVICE: {
+                        _cleanup_free_ char *device = NULL;
+
+                        if (streq(optarg, "auto")) {
+                                if (arg_enroll_type == ENROLL_FIDO2)
+                                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                               "When enrolling a new FIDO2 device, `--unlock-fido2-device=auto` is unsupported. "
+                                                               "Please specify a device to use for unlocking.");
+                        } else {
+                                device = strdup(optarg);
+                                if (!device)
+                                        return log_oom();
+
+                                arg_unlock_fido2_device = TAKE_PTR(device);
```

Comment:
This creates a potential memleak: if `--unlock-fido2-device` is specified more than once, the first `strdup` will be lost. But I think you don't need the strdup at all… Just use a `static const char *` and assing `optarg` directly to it.

Reply:
I kept the `strdup` for consistency with other cases, but I added a guard before all the unlock options just like the enrolling ones to prevent these parameters being passed multiple times.